tests/test_inference_types.py

```diff
@@ -138,3 +140,25 @@ def test_normalize_keys():
     assert isinstance(instance.maybe_items, list)
     assert len(instance.maybe_items) == 1
     assert isinstance(instance.maybe_items[0], DummyType)
+
+
+def test_optional_are_set_to_none():
+    _types = {v for v in types.__dict__.values() if inspect.isclass(v) and issubclass(v, types.BaseInferenceType)}
+    for _type in _types:
+        parameters = inspect.signature(_type).parameters
```

Comment:
This should work as well and is more readable IMO

```suggestion
    for _type in types.BaseInferenceType.__subclasses__:
        parameters = inspect.signature(_type).parameters
```